### ISSUE TITLE
Repository pack file is at .9 GB (total Repo is at 1.1 GB)

### ISSUE THREAD

**Issue by gomker** [nopSolutions/nopCommerce#5762]
The repository seems to have some files in the history that are causing the pack file to become abnormally large. This poses a problem for using the repo in pipeline operations as the excessive size takes some time to download. Using a method I found to parse the pack file [here](https://gist.github.com/gomker/7f808b104adf7a2f4f1518f99916652b) I show the following

````
dlane@hades:.../nopCommerce$ python3 packstats.py
12:00:45 Running: git gc
Enumerating objects: 286421, done.
Counting objects: 100% (286421/286421), done.
Delta compression using up to 8 threads
Compressing objects: 100% (65409/65409), done.
Writing objects: 100% (286421/286421), done.
Total 286421 (delta 214381), reused 286421 (delta 214381)
12:01:06 Running: git verify-pack -v .git/objects/pack/pack-a44976d1a755d7db5550298a59aa86c1bd298b02.idx
12:01:38 Parsing the output
12:01:41 Running: git rev-list --objects --all
12:01:42 Aggregating sizes of files
12:01:44 Done.

Unnormalized:
Cumulat       Size Filename
  2.95%   27302010 src/Presentation/Nop.Web/wwwroot/lib_npm/cldr-data/main/main.zip
  4.78%   16854072 src/Presentation/Nop.Web/App_Data/Install/Fast/create_sample_data.sql
  6.55%   16340038 System.Private.CoreLib.dll
  8.19%   15183735 src/Presentation/Nop.Web/App_Data/Pdf/ARIALUNI.TTF
  9.82%   15096006 src/Presentation/Nop.Web/App_Data/GeoLite2-Country.mmdb
 11.39%   14427361 System.Private.Xml.dll
 12.55%   10804386 src/Presentation/Nop.Web/App_Data/Install/create_sample_data.sql
 13.66%   10222932 coreclr.dll
 14.47%    7483872 src/packages/Microsoft.Data.OData.5.8.2/Microsoft.Data.OData.5.8.2.nupkg
 15.26%    7287143 Microsoft.CodeAnalysis.CSharp.dll
 16.00%    6879379 Nop.Web.dll
 16.73%    6741460 src/Presentation/Nop.Web/wwwroot/lib/cldr-data/main/main.zip
 17.36%    5797981 src/packages/Microsoft.Data.OData.5.8.1/Microsoft.Data.OData.5.8.1.nupkg
 17.99%    5797758 src/packages/Microsoft.Data.OData.5.8.0/Microsoft.Data.OData.5.8.0.nupkg
 18.59%    5580326 src/packages/Microsoft.Data.OData.5.7.0/Microsoft.Data.OData.5.7.0.nupkg
 19.17%    5322201 System.Linq.Expressions.dll
 19.73%    5225470 System.Data.Common.dll
 20.29%    5166691 BouncyCastle.Crypto.dll
 20.84%    5015320 src/packages/EntityFramework.6.0.2/EntityFramework.6.0.2.nupkg
 21.38%    5011115 src/packages/EntityFramework.6.0.1/EntityFramework.6.0.1.nupkg

Normalized:
Cumulat       Size Filename
  3.43%   31732182 *.sql or dump.csv
  6.39%   27302010 src/Presentation/Nop.Web/wwwroot/lib_npm/cldr-data/main/main.zip
  8.16%   16340038 System.Private.CoreLib.dll
  9.80%   15183735 src/Presentation/Nop.Web/App_Data/Pdf/ARIALUNI.TTF
 11.43%   15096006 src/Presentation/Nop.Web/App_Data/GeoLite2-Country.mmdb
 13.00%   14427361 System.Private.Xml.dll
 14.45%   13420932 *.png or *.jpg
 15.55%   10222932 coreclr.dll
 16.36%    7483872 src/packages/Microsoft.Data.OData.5.8.2/Microsoft.Data.OData.5.8.2.nupkg
 17.15%    7287143 Microsoft.CodeAnalysis.CSharp.dll
 17.90%    6879379 Nop.Web.dll
 18.63%    6741460 src/Presentation/Nop.Web/wwwroot/lib/cldr-data/main/main.zip
 19.25%    5797981 src/packages/Microsoft.Data.OData.5.8.1/Microsoft.Data.OData.5.8.1.nupkg
 19.88%    5797758 src/packages/Microsoft.Data.OData.5.8.0/Microsoft.Data.OData.5.8.0.nupkg
 20.49%    5580326 src/packages/Microsoft.Data.OData.5.7.0/Microsoft.Data.OData.5.7.0.nupkg
 21.06%    5322201 System.Linq.Expressions.dll
 21.63%    5225470 System.Data.Common.dll
 22.19%    5166691 BouncyCastle.Crypto.dll
 22.73%    5015320 src/packages/EntityFramework.6.0.2/EntityFramework.6.0.2.nupkg
 23.27%    5011115 src/packages/EntityFramework.6.0.1/EntityFramework.6.0.1.nupkg```

**Comment by RomanovM** [comment]
@gomker And what is your suggestion here, what do you suppose we should do with this?

**Comment by gomker** [comment]
> @gomker And what is your suggestion here, what do you suppose we should do with this?

This would be a your choice as the maintainers to do of course,  I would suggest reviewing these files to see if they can be removed from the commit histrory.  I see some identified as dump files, what appears to be an archived zip file and several packages (nupkg) that I would assume one pull locally rather than being needed in the repository.  I have found sometimes commit history can bloat a repository from old versions of work.

https://docs.github.com/en/github/managing-large-files/working-with-large-files/removing-files-from-a-repositorys-history

**Comment by skoshelev** [comment]
we analyzed a lot of possible actions to reduce the size of the repository. However, not one of the existing methods allows us to save both the history and the working of each of the states. Therefore, we decided to leave everything unchanged. Perhaps we will return to this issue later.

Closed #5762
